modules/ocf_prometheus/manifests/server.pp

```diff
@@ -0,0 +1,152 @@
+class ocf_prometheus::server {
+  include apache
+  include apache::mod::proxy
+  include apache::mod::proxy_http
+  include ocf::firewall::allow_web
+  include ocf::ssl::default
+
+  file {
+    '/usr/local/bin/gen-prometheus-nodes':
+      source => 'puppet:///modules/ocf_prometheus/gen-prometheus-nodes',
+      mode   => '0755';
+  }
+
+  cron { 'gen-prometheus-nodes':
+    command => '/usr/local/bin/gen-prometheus-nodes /var/local/prometheus-nodes.json',
+    minute  => '0',
+    require => File['/usr/local/bin/gen-prometheus-nodes'];
+  }
+
+  exec { 'gen-promethues-nodes-initial':
+    command => '/usr/local/bin/gen-prometheus-nodes /var/local/prometheus-nodes.json',
+    creates => '/var/local/prometheus-nodes.json',
+  }
+
+  file {
+    '/usr/local/bin/gen-prometheus-printers':
+      source => 'puppet:///modules/ocf_prometheus/gen-prometheus-printers',
+      mode   => '0755';
+  }
+
+  cron { 'gen-prometheus-printers':
+    command => '/usr/local/bin/gen-prometheus-printers /var/local/prometheus-printers.json',
+    minute  => '0',
+    require => File['/usr/local/bin/gen-prometheus-printers'];
+  }
+
+  exec { 'gen-prometheus-printers-initial':
+    command => '/usr/local/bin/gen-prometheus-printers /var/local/prometheus-printers.json',
+    creates => '/var/local/prometheus-printers.json',
+  }
+
+  file {
+    '/etc/prometheus/rules.d':
+      ensure  => 'directory',
+      source  => 'puppet:///modules/ocf_prometheus/rules.d',
+      recurse => true;
+  }
+
+  class { '::prometheus::server':
+    version              => '2.4.2',
+    extra_options        => '--web.listen-address="127.0.0.1:9090"',
+    external_url         => 'https://prometheus.ocf.berkeley.edu',
+    rule_files           => [ '/etc/prometheus/rules.d/*' ],
+    alertmanagers_config => [{
+      'scheme'         => 'https',
+      'path_prefix'    => '/alertmanager',
+      'static_configs' => [{
+        'targets' => ['prometheus.ocf.berkeley.edu'],
+      }],
+    }],
+    scrape_configs       => [
+      {
+        'job_name'        => 'prometheus',
+        'scrape_interval' => '10s',
+        'scrape_timeout'  => '10s',
+        'static_configs'  => [
+          {
+            'targets' => [ 'localhost:9090' ],
+            'labels'  => { 'alias' => 'Prometheus' },
+          },
+        ],
+      },
+      {
+        'job_name'        => 'node',
+        'scrape_interval' => '10s',
+        'scrape_timeout'  => '5s',
+
+        'file_sd_configs' => [
+          {
+            'files'            => [ '/var/local/prometheus-nodes.json' ],
+            'refresh_interval' => '1h',
```

Comment:
refresh interval of an hour here, but a cron job that updates the file every minute..?